pkg/cidata/cidata.go

```diff
@@ -50,17 +51,23 @@ func setupEnv(y *limayaml.LimaYAML) (map[string]string, error) {
 			}
 		}
 	}
-	// Replace "localhost" in proxy settings with the gateway address
-	localhostRegexes := []*regexp.Regexp{
-		regexp.MustCompile(`\blocalhost\b`),
-		regexp.MustCompile(`\b127.0.0.1\b`),
-	}
+	// Replace IP that IsLoopback in proxy settings with the gateway address
 	for _, name := range append(lowerVars, upperVars...) {
 		value, ok := env[name]
 		if ok && !strings.EqualFold(name, "no_proxy") {
-			for _, re := range localhostRegexes {
-				value = re.ReplaceAllString(value, qemu.SlirpGateway)
+			u, err := url.Parse(value)
+			if err != nil {
+				logrus.Warnf("Ignoring invalid proxy %q=%v: %s", name, value, err)
+				continue
+			}
+
+			ips, _ := net.LookupIP(u.Hostname())
```

Comment:
Why did you not include error checking here?

Reply:
ips = nil if error != nil, then it will skip the for logic.

expect how to handle the error? warn log? exit? I think not need to handle

Reply:
> ips = nil if error != nil, then it will skip the for logic.

Yes, practically speaking this is correct (looking at the source of `net.LookupIP`), and but from a best practices point of view, you should always check `err`, and ignore all other return values if the value isn't `nil`, as there is no guarantee that the other return values are proper null values or not.

Yes, it is a bit pedantic in this case, but I would prefer to consistently check errors.

> expect how to handle the error? warn log? exit? I think not need to handle

Just don't access the `ips` variable when `err != nil`, e.g.

```go
			if ips, err := net.LookupIP(u.Hostname()); err == nil {
				for _, ip := range ips {
					if ip.IsLoopback() {
						u.Host = net.JoinHostPort(qemu.SlirpGateway, u.Port())
					}
				}
				value = u.String()
			}
```

Reply:
there refactored by the fakeLookupIP, but if not, I would to write
```go
ips, err := net.LookupIP(u.Hostname())
if err != nil {
    logrus.Debugf("net.LookupIP %s: %s", host, err)
    continue
}

for _, ip := range ips {
    newHost := qemu.SlirpGateway
    if u.Port() != "" {
	    newHost = net.JoinHostPort(newHost, u.Port())
    }
    if ip.IsLoopback() {
	    u.Host = newHost
    }
}
```
because we should handle error cases/special conditions first and return early or continue the loop
https://github.com/uber-go/guide/blob/master/style.md#reduce-nesting